data/transactions/verify/txn.go

```diff
@@ -37,13 +42,36 @@ var logicGoodTotal = metrics.MakeCounter(metrics.MetricName{Name: "algod_ledger_
 var logicRejTotal = metrics.MakeCounter(metrics.MetricName{Name: "algod_ledger_logic_rej", Description: "Total transaction scripts executed and rejected"})
 var logicErrTotal = metrics.MakeCounter(metrics.MetricName{Name: "algod_ledger_logic_err", Description: "Total transaction scripts executed and errored"})
 
+// ErrInvalidSignature is the error returned to report that at least one signature is invalid
+var ErrInvalidSignature = errors.New("At least one signature didn't pass verification")
+
+var errSignedTxnHasNoSig = errors.New("signedtxn has no sig")
+var errSignedTxnMaxOneSig = errors.New("signedtxn should only have one of Sig or Msig or LogicSig")
+var errShuttingDownError = errors.New("not verified, verifier is shutting down")
+
 // The PaysetGroups is taking large set of transaction groups and attempt to verify their validity using multiple go-routines.
 // When doing so, it attempts to break these into smaller "worksets" where each workset takes about 2ms of execution time in order
 // to avoid context switching overhead while providing good validation cancelation responsiveness. Each one of these worksets is
 // "populated" with roughly txnPerWorksetThreshold transactions. ( note that the real evaluation time is unknown, but benchmarks
 // show that these are realistic numbers )
 const txnPerWorksetThreshold = 32
 
+// batchSizeBlockLimit is the limit when the batch exceeds, will be added to the exec pool, even if the pool is saturated
+// and the batch verifier will block until the exec pool accepts the batch
+const batchSizeBlockLimit = 512
```

Comment:
(a) Leaving the batch size unbounded may lead to unforeseen effects. I do not know of any such effects, but there is no benefit in having it unbounded. 

(b) The C code handles batches of any size. The benchmarks in `batchverifier_test.go` demonstrate that there is no performance degradation when the sizes grow. 

Combining (a) and (b), `batchSizeBlockLimit` of 512 is an arbitrary bound added which is a safe upper bound for the considered matters.

Reply:
> why not 64 instead of 512

It is chosen by intuition. A multisig transaction can have up to 255 transactions, maybe it is rare, but having a bound less than that won't be interesting. 
It can be a higher value, but I don't have any for or against thoughts about that.